app/Http/Controllers/Api/ClientPublicKeyApiController.php

```diff
@@ -40,12 +51,70 @@ public function __construct
     /**
      * @return array
      */
-    protected function getCreatePayload():array{
-        $payload =  Request::All();
+    protected function getCreatePayload(): array
+    {
+        $payload = Request::All();
         return array_merge($payload, $this->extra_create_payload_params);
     }
 
+    /**
+     * @param int $client_id
+     * @return mixed
+     */
+    public function _getAll($client_id)
```

Comment:
why did u use this private method ?
u need to use the default implementation https://github.com/OpenStackweb/openstackid/blob/5e61b0054c83226b802e4f2f3072087717d8a530/app/Http/Controllers/Traits/GetAllTrait.php#L68 and overload the apply filter